apps/opik-backend/src/main/java/com/comet/opik/domain/filter/FilterQueryBuilder.java

```diff
@@ -42,6 +42,7 @@ public class FilterQueryBuilder {
     private static final String USAGE_PROMPT_TOKENS_ANALYTICS_DB = "usage['prompt_tokens']";
     private static final String USAGE_TOTAL_TOKENS_ANALYTICS_DB = "usage['total_tokens']";
     private static final String VALUE_ANALYTICS_DB = "value";
+    private static final String DURATION_ANALYTICS_DB = "(date_diff('microsecond', start_time, end_time) / 1000.0)";
```

Comment:
This should be moved to the query level on the DAO, for both spans and traces. 

The reason is for a good separation of responsibilities, as filters shouldn't be calculating data. They should just take care of filtering.

Additionally, it isn't clear why dividing by 1000. It's inconsistent with the existing calculation code. Please se existing code as reference, example:

`if(end_time IS NOT NULL, date_diff('microsecond', start_time, end_time), null) as duration`

Reply:
Agree. I will use some of @thiagohora's changes in the later PR to use a materialized column instead.
As for the division by 1000, I believe the idea is to have a millisecond value with microsecond precision. If `date_diff('millisecond', ...)` is used than the result will be `0` for durations smaller than a millisecond.
As for why microsecond is used and not nanosecond, from what I see, `date_diff` support for `nanosecond` is not supported in the version we are using. [Support for this introduced](https://github.com/ClickHouse/ClickHouse/commit/5020741b253f66967ef84707c98a2fd201d5080d) several months ago.

Reply:
Done in https://github.com/comet-ml/opik/pull/883/commits/d7ad75042780b89ed58b8b768e187423e2f38cca